File: languages/python/bloock/entity/authenticity/signature_header.py
from __future__ import annotations

import bloock._bridge.proto.authenticity_entities_pb2 as proto


class SignatureHeader:
    def __init__(self, alg: str, kid: str) -> None:
        self.alg = alg
        self.kid = kid

    @staticmethod
    def from_proto(header: proto.SignatureHeader) -> SignatureHeader:
        return SignatureHeader(alg=header.alg, kid=header.kid)

    def to_proto(self) -> proto.SignatureHeader:
        return proto.SignatureHeader(alg=self.alg, kid=self.kid)

File: languages/php/Bloock/Entity/Authenticity/SignatureHeader.php
<?php

namespace Bloock\Entity\Authenticity;

class SignatureHeader
{
    private string $alg;
    private string $kid;

    public function __construct(string $alg, string $kid)
    {
        $this->alg = $alg;
        $this->kid = $kid;
    }

    public static function fromProto(\Bloock\SignatureHeader $header): SignatureHeader
    {
        return new SignatureHeader($header->getAlg(), $header->getKid());
    }

    /**
     * @return string
     */
    public function getAlg(): string
    {
        return $this->alg;
    }

    /**
     * @return string
     */
    public function getKid(): string
    {
        return $this->kid;
    }

    public function toProto(): \Bloock\SignatureHeader
    {
        $p = new \Bloock\SignatureHeader();
        $p->setAlg($this->alg);
        $p->setKid($this->kid);
        return $p;
    }
}